src/App.tsx

```diff
@@ -17,18 +43,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                setFilter={setFilter}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {!isLoading && todos.length > 0 && (
+                <TodoList
```

Comment:
I think, will be correctly like this
```suggestion
              {isLoading ? (
                <Loader />
              ) : (
                <TodoList ...
```